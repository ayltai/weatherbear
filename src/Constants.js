export const Constants = {};

Constants.TYPE_DARK_SKY     = 0;
Constants.TYPE_ACCU_WEATHER = 1;
Constants.TYPE_WEATHER_BIT  = 2;

Constants.UNIT_SI       = 'si';
Constants.UNIT_IMPERIAL = 'imperial';
