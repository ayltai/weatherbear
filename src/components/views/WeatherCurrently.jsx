import { Grid, makeStyles, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

import { IconHelper, } from '../../utils/IconHelper';
import { NumberFormatHelper, } from '../../utils/NumberFormatHelper';
import { Constants, } from '../../Constants';
import { ShadowedText, } from './ShadowedText';
import { Temperature, } from './Temperature';

export const WeatherCurrently = props => {
    const classes = makeStyles(theme => ({
        background : {
            backgroundImage : `url(${props.backgroundImageUrl})`,
            backgroundSize  : 'cover',
            padding         : theme.spacing(0.5),
            height          : props.height + theme.spacing(1),
        },
        padding    : {
            paddingTop : theme.spacing(1),
        },
    }))();

    const humidity = NumberFormatHelper.toFixed(props.humidity * 100, 0);

    return (
        <div className={classes.background}>
            <ShadowedText
                variant='body2'
                tooltip={props.location}
                align='center'
                noShadow={props.noShadow}
                noWrap>
                {props.location}
            </ShadowedText>
            <ShadowedText
                variant='body2'
                tooltip={props.summaryCurrently}
                align='center'
                noShadow={props.noShadow}
                noWrap>
                {props.summaryCurrently}
            </ShadowedText>
            <Grid
                container
                alignItems='center'
                justify='space-around'>
                <Grid
                    item
                    xs={4}>
                    <Temperature
                        variant='h4'
                        align='center'
                        unit={props.unit}
                        noShadow={props.noShadow}
                        showTip
                        noWrap>
                        {props.temperature}
                    </Temperature>
                    <ShadowedText
                        tooltip={`${props.humidityLabel || 'Humidity'}: ${humidity}%`}
                        variant='body2'
                        align='center'
                        noShadow={props.noShadow}
                        noWrap>
                        {`ðŸ’§ ${humidity}%`}
                    </ShadowedText>
                </Grid>
                <Grid
                    item
                    xs={4}>
                    <ShadowedText
                        tooltip={`${props.windSpeedLabel || 'Wind'}: ${props.windSpeed}, ${props.uvIndexLabel || 'UV'}: ${NumberFormatHelper.toFixed(props.uvIndex, 0)}`}
                        variant='h2'
                        align='center'
                        noShadow={props.noShadow}>
                        <i className={`wi ${IconHelper.toIcon(props.iconId, props.iconType || Constants.TYPE_DARK_SKY)}`}></i>
                    </ShadowedText>
                </Grid>
                <Grid
                    item
                    xs={4}>
                    <Temperature
                        variant='body2'
                        align='center'
                        prefix={`${props.temperatureHighLabel || 'ðŸ¥µ'} `}
                        unit={props.unit}
                        noShadow={props.noShadow}
                        showTip
                        noWrap>
                        {props.temperatureHigh}
                    </Temperature>
                    <Temperature
                        variant='body2'
                        align='center'
                        prefix={`${props.temperatureLowLabel || 'ðŸ¥¶'} `}
                        unit={props.unit}
                        noShadow={props.noShadow}
                        showTip
                        noWrap>
                        {props.temperatureLow}
                    </Temperature>
                </Grid>
            </Grid>
            <ShadowedText
                variant='caption'
                tooltip={props.summaryToday}
                align='center'
                noShadow={props.noShadow}
                noWrap>
                {props.summaryToday}
            </ShadowedText>
        </div>
    );
};

WeatherCurrently.propTypes = {
    width                : PropTypes.number.isRequired,
    height               : PropTypes.number.isRequired,
    backgroundImageUrl   : PropTypes.string,
    location             : PropTypes.string,
    summaryCurrently     : PropTypes.string,
    summaryToday         : PropTypes.string,
    iconId               : PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]),
    iconType             : PropTypes.oneOf([
        Constants.TYPE_ACCU_WEATHER,
        Constants.TYPE_DARK_SKY,
        Constants.TYPE_WEATHER_BIT,
    ]),
    temperature          : PropTypes.number,
    temperatureHigh      : PropTypes.number,
    temperatureHighLabel : PropTypes.string,
    temperatureLow       : PropTypes.number,
    temperatureLowLabel  : PropTypes.string,
    humidity             : PropTypes.number,
    humidityLabel        : PropTypes.string,
    windSpeed            : PropTypes.number,
    windSpeedLabel       : PropTypes.string,
    uvIndex              : PropTypes.number,
    uvIndexLabel         : PropTypes.string,
    unit                 : PropTypes.oneOf([
        Constants.UNIT_SI,
        Constants.UNIT_IMPERIAL,
    ]),
    noShadow             : PropTypes.bool,
};
