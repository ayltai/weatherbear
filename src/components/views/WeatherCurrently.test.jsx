import { mount, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { Constants, } from '../../Constants';
import { WeatherCurrently, } from './WeatherCurrently';

const width            = 200;
const height           = 100;
const location         = 'Los Angeles';
const summaryCurrently = 'Partly Cloudy';
const summaryToday     = 'No precipitation throughout the week';
const iconId           = 'partly-cloudy-night';
const iconType         = Constants.TYPE_DARK_SKY;
const temperature      = 28.3;
const temperatureHigh  = 29.7;
const temperatureLow   = 22.4;
const humidity         = 0.77;
const windSpeed        = 7;
const uvIndex          = 4;
const unit             = Constants.UNIT_SI;

const component = (
    <WeatherCurrently
        width={width}
        height={height}
        location={location}
        summaryCurrently={summaryCurrently}
        summaryToday={summaryToday}
        iconId={iconId}
        iconType={iconType}
        temperature={temperature}
        temperatureHigh={temperatureHigh}
        temperatureLow={temperatureLow}
        humidity={humidity}
        windSpeed={windSpeed}
        uvIndex={uvIndex}
        unit={unit} />
);

describe('<WeatherCurrently />', () => {
    it('renders correctly', () => {
        expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(component);
    });
});
