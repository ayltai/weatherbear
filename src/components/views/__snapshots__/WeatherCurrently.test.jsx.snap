// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherCurrently /> renders correctly 1`] = `
<div>
  <canvas
    className="makeStyles-canvas-1"
    height={100}
    id="canvas"
    width={200}
  />
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-3 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title="Los Angeles"
  >
    Los Angeles
  </p>
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-5 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title="Partly Cloudy"
  >
    Partly Cloudy
  </p>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <p
        aria-describedby={null}
        className="MuiTypography-root makeStyles-textShadowed-7 MuiTypography-h4 MuiTypography-noWrap MuiTypography-alignCenter"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="28.3Â°C"
      >
        28Â°C
      </p>
      <p
        aria-describedby={null}
        className="MuiTypography-root makeStyles-textShadowed-9 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Humidity: 77%"
      >
        ðŸ’§ 77%
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <p
        aria-describedby={null}
        className="MuiTypography-root makeStyles-textShadowed-11 MuiTypography-h2 MuiTypography-alignCenter"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Wind: 7, UV: 4"
      >
        <i
          className="wi wi-night-alt-cloudy"
        />
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <p
        aria-describedby={null}
        className="MuiTypography-root makeStyles-textShadowed-13 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="29.7Â°C"
      >
        ðŸ¥µ 30Â°C
      </p>
      <p
        aria-describedby={null}
        className="MuiTypography-root makeStyles-textShadowed-15 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="22.4Â°C"
      >
        ðŸ¥¶ 22Â°C
      </p>
    </div>
  </div>
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-17 MuiTypography-caption MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title="No precipitation throughout the week"
  >
    No precipitation throughout the week
  </p>
</div>
`;
